public/components/common/seach/queryBar.tsx

```diff
@@ -9,28 +9,137 @@
  * GitHub history for details.
  */
 
-import React from 'react';
+import React, {Fragment, useEffect, useState, useCallback} from 'react';
 import {
+  EuiButtonEmpty,
   EuiFieldSearch,
   EuiFlexItem,
+  EuiSuggest,
+  EuiFlexGroup,
+  EuiSelectableTemplateSitewide,
+  EuiSelectableTemplateSitewideOption,
+  EuiSuggestItem,
+  EuiSuggestItemProps
 } from '@elastic/eui';
 import {
   IQueryBarProps
 } from './search';
 import {
   RAW_QUERY
 } from '../../../common/constants/explorer';
+import { Test } from 'mocha';
 
-export function QueryBar(props: IQueryBarProps) {
+const commands: EuiSuggestItemProps[] =[
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'dedup'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'eval'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'fields'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'rename'},
+  //{type: {iconType: 'search', color: 'tint1'} ,label: 'search'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'sort'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'stats'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'where'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'head'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'top'},
+  {type: {iconType: 'search', color: 'tint1'}, label: 'source'}
+]
+const firstCommand: EuiSuggestItemProps[] = [
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'search'},
+  {type: {iconType: 'search', color: 'tint1'}, label: 'source'}
+]
+const pipeCommands: EuiSuggestItemProps[] = [
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'dedup'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'eval'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'fields'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'rename'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'sort'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'stats'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'where'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'head'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'top'},
+]
 
+const indices: EuiSuggestItemProps[] = [
+  {type: {iconType: 'kqlField', color: 'tint4'} ,label: 'opensearch_dashboards_sample_data_ecommerce'},
+  {type: {iconType: 'kqlField', color: 'tint4'} ,label: 'opensearch_dashboards_sample_data_flights'},
+  {type: {iconType: 'kqlField', color: 'tint4'} ,label: 'opensearch_dashboards_sample_data_logs'}
+]
+//
+export function QueryBar(props: IQueryBarProps) {
+  
   const {
     query,
     handleQueryChange,
     handleQuerySearch
   } = props;
 
+  const [inputValue, setInputValue] = useState('');
+
+  const splittedModel = inputValue.split(' ');
+  const prefix = splittedModel[splittedModel.length - 1];
+  // const suggestions = commands.filter(
```

Comment:
Are all these commented-out code blocks safe to remove?

Reply:
Yup, they were just leftover from previous commits.

Reply:
Got you- please remove these in your next commit then